tests/components/dyson/test_climate.py

```diff
@@ -154,236 +136,259 @@ def _get_device_heat_on():
+async def test_pure_hot_cool_link_state(mocked_login, mocked_devices, hass):
+    """Test set climate temperature."""
+    await async_setup_component(hass, DYSON_DOMAIN, _get_config())
+    await hass.async_block_till_done()
+
+    state = hass.states.get("climate.temp_name")
+    assert state.attributes[ATTR_SUPPORTED_FEATURES] == SUPPORT_FLAGS
+    assert state.attributes[ATTR_TEMPERATURE] == 23
+    assert state.attributes[ATTR_CURRENT_TEMPERATURE] == 289 - 273
+    assert state.attributes[ATTR_CURRENT_HUMIDITY] == 53
+    assert state.state == HVAC_MODE_HEAT
+    assert len(state.attributes[ATTR_HVAC_MODES]) == 2
+    assert HVAC_MODE_HEAT in state.attributes[ATTR_HVAC_MODES]
+    assert HVAC_MODE_COOL in state.attributes[ATTR_HVAC_MODES]
+    assert len(state.attributes[ATTR_FAN_MODES]) == 2
+    assert FAN_FOCUS in state.attributes[ATTR_FAN_MODES]
+    assert FAN_DIFFUSE in state.attributes[ATTR_FAN_MODES]
+
+    device = mocked_devices.return_value[0]
+    update_callback = device.add_message_listener.call_args[0][0]
+
+    device.state.focus_mode = FocusMode.FOCUS_ON.value
+    await hass.async_add_executor_job(update_callback, MockDysonState())
+    await hass.async_block_till_done()
+
+    state = hass.states.get("climate.temp_name")
+    assert state.attributes[ATTR_FAN_MODE] == FAN_FOCUS
+
+    device.state.focus_mode = FocusMode.FOCUS_OFF.value
+    await hass.async_add_executor_job(update_callback, MockDysonState())
+    await hass.async_block_till_done()
+
+    state = hass.states.get("climate.temp_name")
+    assert state.attributes[ATTR_FAN_MODE] == FAN_DIFFUSE
+
+    device.state.heat_mode = HeatMode.HEAT_ON.value
+    device.state.heat_state = HeatState.HEAT_STATE_OFF.value
+    await hass.async_add_executor_job(update_callback, MockDysonState())
+    await hass.async_block_till_done()
+
+    state = hass.states.get("climate.temp_name")
+    assert state.state == HVAC_MODE_HEAT
+    assert state.attributes[ATTR_HVAC_ACTION] == CURRENT_HVAC_IDLE
+
+    device.environmental_state.humidity = 0
+    await hass.async_add_executor_job(update_callback, MockDysonState())
+    await hass.async_block_till_done()
+
+    state = hass.states.get("climate.temp_name")
+    assert state.attributes.get(ATTR_CURRENT_HUMIDITY) is None
+
+    device.environmental_state = None
+    await hass.async_add_executor_job(update_callback, MockDysonState())
+    await hass.async_block_till_done()
+
+    state = hass.states.get("climate.temp_name")
+    assert state.attributes.get(ATTR_CURRENT_HUMIDITY) is None
+
+    device.state.heat_mode = HeatMode.HEAT_OFF.value
+    device.state.heat_state = HeatState.HEAT_STATE_OFF.value
+    await hass.async_add_executor_job(update_callback, MockDysonState())
+    await hass.async_block_till_done()
+
+    state = hass.states.get("climate.temp_name")
+    assert state.state == HVAC_MODE_COOL
+    assert state.attributes[ATTR_HVAC_ACTION] == CURRENT_HVAC_COOL
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6184085)